### PR TITLE
Fix passing _ddp_params_and_buffers_to_ignore

### LINKED ISSUE #11844: _LightningModuleWrapperBase doesn't pass attribute _ddp_params_and_buffers_to_ignore correctly
## 🐛 Bug

I guess, there's a problem when passing `_ddp_params_and_buffers_to_ignore` attribute to the wrapper. There's the corresponding section - https://github.com/PyTorchLightning/pytorch-lightning/blob/master/pytorch_lightning/overrides/base.py#L75 .

I think it should be prefixed with the name of attribute where it is saved.

```diff
class _LightningModuleWrapperBase(DeviceDtypeModuleMixin, torch.nn.Module):
    def __init__(self, pl_module: Union["pl.LightningModule", _LightningPrecisionModuleWrapperBase]):
        """
        Wraps the user's LightningModule and redirects the forward call to the appropriate
        method, either ``training_step``, ``validation_step`` or ``test_step``.
        If the LightningModule is in none of the states `training`, `testing` or `validation`,
        the inputs will be redirected to the
        :meth:`~pytorch_lightning.core.lightning.LightningModule.predict` method.
        Inheriting classes may also modify the inputs or outputs of forward.

        Args:
            pl_module: the model to wrap
        """
        super().__init__()
        self.module = pl_module

        # set the parameters_to_ignore from LightningModule.
-        self._ddp_params_and_buffers_to_ignore = getattr(pl_module, "_ddp_params_and_buffers_to_ignore", [])
+        self._ddp_params_and_buffers_to_ignore = list(map(lamba x: f"module.{x}", getattr(pl_module, "_ddp_params_and_buffers_to_ignore", [])))
```
### To Reproduce

You can set anything to `self._ddp_params_and_buffers_to_ignore` in `LightningModule` and then print the values in here - https://github.com/PyTorchLightning/pytorch-lightning/blob/9ebdc52ec631df92ca7cfc1ba852801dd36d3864/pytorch_lightning/plugins/training_type/ddp.py#L188 .
It will not have the prefixes of `module` caused by the wrapping, leading to situation where the params to ignore are not correctly passed to the `DistributedDataParallel`, which then can cause errors.

### Expected behavior

Parameter names should be prefixed with the modules while wrapping. This will fix later errors when parameters are not ignored properly.

### Environment

* CUDA:
        - GPU:
        - available:         False
        - version:           11.1
* Packages:
        - numpy:             1.22.1
        - pyTorch_debug:     False
        - pyTorch_version:   1.10.0+cu111
        - pytorch-lightning: 1.5.8
        - tqdm:              4.62.3
* System:
        - OS:                Linux
        - architecture:
                - 64bit
                - 
        - processor:         x86_64
        - python:            3.8.0
        - version:           #1 SMP Fri Sep 10 13:19:36 EDT 2021

### Additional context

cc @tchaton @rohitgr7 @akihironitta @carmocca @justusschock @awaelchli @borda @ananthsub @ninginthecloud @jjenniferdai @kaushikb11

### PR DESCRIPTION
**Pull request by Honzys**
## What does this PR do?

When wrapping the LightningModule with some wrapper
classes, names of the params and buffers to ignore
should be prefixed with the name of the variable to which
is the LightningModule assigned.

Fixes #11844

### Does your PR introduce any breaking changes? If yes, please list them.

I hope not.

## Before submitting

- [x] Was this **discussed/approved** via a GitHub issue? (not for typos and docs)
- [x] Did you read the [contributor guideline](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/.github/CONTRIBUTING.md), **Pull Request** section?
- [x] Did you make sure your **PR does only one thing**, instead of bundling different changes together?
- [x] Did you make sure to **update the documentation** with your changes? (if necessary)
- [x] Did you write any **new necessary tests**? (not for typos and docs)
- [x] Did you verify new and **existing tests pass** locally with your changes?
- [x] Did you list all the **breaking changes** introduced by this pull request?
- [x] Did you **update the [CHANGELOG](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/CHANGELOG.md)**? (not for typos, docs, test updates, or internal minor changes/refactorings)

## PR review

Anyone in the community is welcome to review the PR.
Before you start reviewing make sure you have read [Review guidelines](https://github.com/PyTorchLightning/pytorch-lightning/wiki/Review-guidelines). In short, see the following bullet-list:

- [x] Is this pull request ready for review? (if not, please submit in draft mode)
- [x] Check that all items from **Before submitting** are resolved
- [x] Make sure the title is self-explanatory and the description concisely explains the PR
- [x] Add labels and milestones (and optionally projects) to the PR so it can be classified

## Did you have fun?

Make sure you had fun coding 🙃

### CODE REVIEW

**Comment by Honzys** [comment]
I guess I should update the CHANGELOG -> should I add the comment under the version `1.6.0` or somewhere else?
Also should I change the target branch to `release/1.5.x` as the milestone added to the issue is `1.5.x`?

**Comment by awaelchli** [comment]
@Honzys nice

You can add the changelog under the 1.6.0 "Fixed" section. Your branch should target master like, no further action required :)

**Review by awaelchli — APPROVED** [review]
Solid fix!

**Review by akihironitta — APPROVED** [review]
Nice catch! Once the changelog gets updated, LGTM!

**Comment by Honzys** [comment]
I've just updated the `CHANGELOG`, so it should be ready from my part.

**Review by kaushikb11 — APPROVED** [review]

**Review by rohitgr7 — APPROVED** [review]

**Review by carmocca — APPROVED** [review]

**Review by tchaton — APPROVED** [review]
LGTM !
